src/traccuracy/track_errors/divisions.py

```diff
@@ -165,8 +164,17 @@ def _correct_shifted_divisions(matched_data: Matched, n_frames: int = 1) -> Matc
     # Compare all pairs of fp and fn
     for fp_node, fn_node in itertools.product(fp_divs, fn_divs):
         correct = False
-        t_fp = g_pred.graph.nodes[fp_node][g_pred.frame_key]
-        t_fn = g_gt.graph.nodes[fn_node][g_gt.frame_key]
+        fp_node_info = g_pred.graph.nodes[fp_node]
+        fn_node_info = g_gt.graph.nodes[fn_node]
+        t_fp = fp_node_info[g_pred.frame_key]
+        t_fn = fn_node_info[g_gt.frame_key]
+
+        # Move on if this division has already been corrected by a smaller buffer value
+        if (
+            fp_node_info.get("min_buffer_correct", np.nan) is not np.nan
+            and fn_node_info.get("min_buffer_correct", np.nan) is not np.nan
```

Comment:
Should this be an or? E.g. if one of them has a non nan value that means that it's already been matched to another division that can be corrected with a frame shift.

Reply:
Yep, good catch!